api/app/Enums/NotificationType.php

```diff
@@ -10,6 +10,7 @@
 enum NotificationType
```

Comment:
I think this enum is redundant. I switched the type discriminator to use the built-in __typename graphql property. The Laravel Notification system automatically creates a type column for itself to use, too.

Reply:
I'm kinda confused by this. There is another NotificationType in the front end which has been used to define email or in-app notifications (https://github.com/GCTC-NTGC/gc-digital-talent/blob/1c1419cf935dd13158248b3f90c81acd9e99a225/apps/web/src/pages/Profile/AccountSettings/types.ts#L8) . If you don't need this can we get rid of it?

Reply:
@esizer Are you OK if I remove this enum?  I think you added it just for type discrimination right?

Reply:
Uhh, that was actually @yonikid15 who added it think :sweat_smile: I am fine with it being removed though :stuck_out_tongue:

Reply:
Cool, thanks.  :grin: 

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/540134ba-86fe-406d-90a0-de7736b04f29)

Reply:
[remove enum](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10230/commits/c19e52c3ddaf614dd67b5bf4aa7c3f8e9fa87cc7) and [remove enum](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10230/commits/25cb849336ef93556f618e397cc688c158e1c7d6)